### ISSUE TITLE
Move amo-validator calling to subprocess

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-server#9670]
As part of #9105 and an unsure path forward, we should just move to call the amo-validator to a subprocess to make it easier to ignore it for future work.

**Comment by diox** [comment]
Random thought: maybe it will make things harder, so we gotta be careful here. Having it as a subprocess doesn't solve dependencies issues - it allows us to have python 2 and python 3 side by side more easily yes, but while we're on python 2 the same env will be used for addons-server and amo-validator, so we can't just blindly upgrade dependencies that both projects depend on. For instance, I [filed an issue about upgrading `rdflib`](https://github.com/mozilla/addons-server/issues/9701), and I thought I'd wait until this issue is closed as it'd make my life easier, but it's not actually the case: both need to work with the newer version.

**Comment by EnTeQuAk** [comment]
This issue is based on the fact that the validator is primarily used for now… 2 or 3 internally signed mozilla extensions of which all are in the process of being ported from what I remember from recent IRC discussions. Given that we'll be able to deactivate the validator rather sooner than later (I hope… sincerely) and with the change here simply dropping it got a lot easier.

In any case, I think I disagree. The validator can be linked specifically to a python 2 environment, even in the same docker file. Once we have python 2 and 3 running side by side we will have to install dependencies for both envs in parallel anyway so we can also quite freely decide to link the validator directly against the python 2 environment and the binary can then be called from the python 3 web-env via subprocess.

Unless I misunderstood you?

**Comment by diox** [comment]
> I think this doesn't have to be the case. For an interim period we could built both, python 2 and 3 in parallel and install the same dependencies and switch between python 2 and 3 based on an ops controlled environment variable to slowly progress. This way the same docker image can run both and with that the validator linked to the python 2 env.

The problem I identified was *only* when both addons-server and amo-validator are running in python 2 env.

**Comment by EnTeQuAk** [comment]
> The problem I identified was only when both addons-server and amo-validator are running in python 2 env.

Eh, right, this whole refactoring doesn't touch any of the actual dependencies in our current environment or simplifies the situation in that regard. I misread your statement I guess. I don't think it makes it worse though I think :thinking:

**Comment by vcarciu** [comment]
@EnTeQuAk : @EnTeQuAk : Please add some STRs to this issue or mark it as "qa: not needed"

**Comment by EnTeQuAk** [comment]
Alrighty, chatted with @vcarciu on IRC about this and tested it on -dev:

* Submitted a variation of the cliqz add-on (removed a lot of files to avoid running into timeout issues) on -dev (keeping META-INF and most of the rdf intact)
* Received an akismet spam checking error
 * BUT these are run *after* amo-validator so the validator ran successfully
 * This is the validation report: https://addons-dev.allizom.org/en-US/developers/upload/e3a9ad768505474fa539002f9f2bb83e
  * The "Localization Tests" prove that the validator actually ran, this result is thrown by the validator and not some addons-server code

With that we're marking this as verified fixed.
